crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -363,107 +244,295 @@ impl Object {
+        ObjectImpl(self).version().await
+    }
+
+    /// 32-byte hash that identifies the object's current contents, encoded as a Base58 string.
+    pub(crate) async fn digest(&self) -> String {
+        ObjectImpl(self).digest().await
+    }
+
+    /// The owner type of this object: Immutable, Shared, Parent, Address
+    /// Immutable and Shared Objects do not have owners.
+    pub(crate) async fn owner(&self, ctx: &Context<'_>) -> Option<ObjectOwner> {
+        ObjectImpl(self).owner(ctx).await
+    }
+
+    /// The transaction block that created this version of the object.
+    pub(crate) async fn previous_transaction_block(
+        &self,
+        ctx: &Context<'_>,
+    ) -> Result<Option<TransactionBlock>> {
+        ObjectImpl(self).previous_transaction_block(ctx).await
+    }
+
+    /// The amount of SUI we would rebate if this object gets deleted or mutated. This number is
+    /// recalculated based on the present storage gas price.
+    pub(crate) async fn storage_rebate(&self) -> Option<BigInt> {
+        ObjectImpl(self).storage_rebate().await
     }
 
-    /// Access a dynamic field on an object using its name.
-    /// Names are arbitrary Move values whose type have `copy`, `drop`, and `store`, and are specified
-    /// using their type, and their BCS contents, Base64 encoded.
-    /// Dynamic fields on wrapped objects can be accessed by using the same API under the Owner type.
-    pub async fn dynamic_field(
+    /// The transaction blocks that sent objects to this object.
+    pub(crate) async fn received_transaction_blocks(
+        &self,
+        ctx: &Context<'_>,
+        first: Option<u64>,
+        after: Option<transaction_block::Cursor>,
+        last: Option<u64>,
+        before: Option<transaction_block::Cursor>,
+        filter: Option<TransactionBlockFilter>,
+    ) -> Result<Connection<String, TransactionBlock>> {
+        ObjectImpl(self)
+            .received_transaction_blocks(ctx, first, after, last, before, filter)
+            .await
+    }
+
+    /// The Base64-encoded BCS serialization of the object's content.
+    pub(crate) async fn bcs(&self) -> Result<Option<Base64>> {
+        ObjectImpl(self).bcs().await
+    }
+
+    /// The set of named templates defined on-chain for the type of this object, to be handled
+    /// off-chain. The server substitutes data from the object into these templates to generate a
+    /// display string per template.
+    async fn display(&self, ctx: &Context<'_>) -> Result<Option<Vec<DisplayEntry>>> {
+        ObjectImpl(self).display(ctx).await
+    }
+
+    /// Access a dynamic field on an object using its name. Names are arbitrary Move values whose
+    /// type have `copy`, `drop`, and `store`, and are specified using their type, and their BCS
+    /// contents, Base64 encoded.
+    ///
+    /// Dynamic fields on wrapped objects can be accessed by using the same API under the Owner
+    /// type.
+    async fn dynamic_field(
```

Comment:
Forgive me for being a little slow, but why are some of these fns pub(crate) and others not?

Reply:
iiuc it's because this is not intended to be an ObjectImpl usable by other Objects. Whether it's pub(crate) or private does not matter to async-graphql

Reply:
Yeah it's not super obvious why this is (I originally went through and made everything private, but then `async-graphql` quickly corrected me) -- functions that correspond to interface implementations need to be non-private (this makes sense because `async-graphql` needs to write forwarding functions on the `enum` that represents the interface to these functions).